docs/data-sources/stacks.md

```diff
@@ -0,0 +1,40 @@
+# Stack Data Source
+
+[Stacks](https://deploy-docs.aptible.com/docs/stacks)
+are the underlying virtualized infrastructure (EC2 instances, private network, etc.) your resources
+are deployed on.
+
+## Example Usage
+
+### Determining the Stack ID
+
+If you have an Stack with the name "test-stack" you can
+create the data source:
+
+```hcl
+data "aptible_stack" "test-stack" {
+    name = "test-stack"
+}
+```
+
+Once defined, you can use this data source in your resource definitions.
+For example, when defining an environment:
+
+```hcl
+resource "aptible_environment" "test-env" {
+    env_id = data.aptible_stack.test-stack.stack_id
+    name = "test-env"
+}
```

Comment:
Should be `stack_id` not `env_id`. Also, Environment resources currently require an organization ID. Assuming that the stack is a dedicated Stack, the `org_id` can be retrieved from it.

```suggestion
resource "aptible_environment" "test-env" {
    stack_id = data.aptible_stack.test-stack.stack_id
    org_id = data.aptible_stack.test-stack.org_id
    name = "test-env"
}
```